packages/api/cms-api/src/dependencies/entities/dependency.entity.ts

```diff
@@ -1,7 +1,9 @@
 import { Field, ObjectType } from "@nestjs/graphql";
 
+// Dependency is not an actual entity because it represents a view
+// and view entites are not supported yet https://github.com/mikro-orm/mikro-orm/issues/672#issuecomment-660877928
```

Comment:
for what reason did you name it entitiy then?

Reply:
Because it still represents a row in the database (just a row of a view, not a table). 

I needed a way to distinguish the dependency saved in the DB (`DependencyEntity`) and the dependency returned the DependenciesService (`Dependency`, also includes the `name` and `secondaryInformation` from `@EntityInfo()`).

Reply:
I understand, tricky.

Ideas:

- create just one class and make name/secondaryInfo optional
- or, name it Dependency and DependencyDto
- or, turn the first into a interface named DependencyInterface or so and the second Dependency (a class)

Reply:
> create just one class and make name/secondaryInfo optional

They're already optional, so this should be okay

Reply:
You could also call it BaseDependency

Reply:
Could the BaseDependency also be just an interface? (because it is not an `@ObjectType` it doesn't have to be a class, and the `@Field()` decorators you can remove as they have no effect here)

Reply:
True, I converted it to be an interface